python-checks/src/test/resources/checks/pytorch_estimator_hyperparameters.py

```diff
@@ -0,0 +1,157 @@
+    from torch.optim import NAdam
+    noncompliant = NAdam() # Noncompliant {{Add the missing hyperparameters lr, weight_decay and momentum_decay for this PyTorch optimizer.}}
+    noncompliant = NAdam(model.parameters(), lr = 0.001) # Noncompliant
+    noncompliant = NAdam(model.parameters(), weight_decay = 0.001) # Noncompliant
+    noncompliant = NAdam(model.parameters(), momentum_decay = 0.001) # Noncompliant
+
+    optimizer = NAdam(model.parameters(), 0.001, betas, eps, 0.23, 0.25)
+    optimizer = NAdam(model.parameters(), lr = 0.001, weight_decay=0.23, momentum_decay=0.25)
+    optimizer = NAdam(model.parameters(), **param_dict, *param_list)
+    optimizer = NAdam(model.parameters(), **param_dict, *param_list)
+
+def radam(param_dict, param_list, betas, eps):
+    from torch.optim import RAdam
+    noncompliant = RAdam() # Noncompliant
+    noncompliant = RAdam(model.parameters(), lr = 0.001) # Noncompliant
+    noncompliant = RAdam(model.parameters(), weight_decay = 0.001) # Noncompliant
+
+    optimizer = RAdam(model.parameters(), 0.001, betas, eps, 0.23)
+    optimizer = RAdam(model.parameters(), lr = 0.001, weight_decay=0.23)
+    optimizer = RAdam(model.parameters(), **param_dict, *param_list)
+    optimizer = RAdam(model.parameters(), **param_dict, *param_list)
+
+def rms_prop(param_dict, param_list, alpha, eps):
+    from torch.optim import RMSprop
+    noncompliant = RMSprop() # Noncompliant
+    noncompliant = RMSprop(model.parameters(), lr = 0.001) # Noncompliant
+    noncompliant = RMSprop(model.parameters(), weight_decay = 0.001) # Noncompliant
+    noncompliant = RMSprop(model.parameters(), momentum = 0.001) # Noncompliant
+
+    optimizer = RMSprop(model.parameters(), 0.001, alpha, eps, 0.23, 0.323)
+    optimizer = RMSprop(model.parameters(), lr = 0.001, weight_decay=0.23, momentum=0.323)
+    optimizer = RMSprop(model.parameters(), **param_dict, *param_list)
+    optimizer = RMSprop(model.parameters(), **param_dict, *param_list)
+
+def rprop(param_dict, param_list):
+    from torch.optim import Rprop
+    noncompliant = Rprop() # Noncompliant
+    noncompliant = Rprop(model.parameters()) # Noncompliant
+
+    optimizer = Rprop(model.parameters(), 0.001)
+    optimizer = Rprop(model.parameters(), lr = 0.001)
+    optimizer = Rprop(model.parameters(), **param_dict, *param_list)
+    optimizer = Rprop(model.parameters(), **param_dict, *param_list)
+
+def sgd(param_dict, param_list, dampening):
+    from torch.optim import SGD
+    noncompliant = SGD() # Noncompliant
+    noncompliant = SGD(model.parameters(), lr = 0.001) # Noncompliant
+    noncompliant = SGD(model.parameters(), weight_decay = 0.001) # Noncompliant
+    noncompliant = SGD(model.parameters(), momentum = 0.001) # Noncompliant
+
+    optimizer = SGD(model.parameters(), 0.001, 0.323, dampening, 0.23)
+    optimizer = SGD(model.parameters(), lr = 0.001, momentum=0.323, weight_decay=0.23)
+    optimizer = SGD(model.parameters(), **param_dict, *param_list)
```

Comment:
I am not sure why but the last two lines of each function are duplicated. Also I would suggest not to test multiple times the same thing.  For example on line 156 your are testing that we do not raise on spread operator, but this is already tested on line 145.